src/audio/tdfb/tdfb_direction.c

```diff
@@ -159,7 +157,8 @@ static bool line_array_mode_check(struct tdfb_comp_data *cd)
 	 * Form vector AB(a,b,c) from x(i+1) - x(i), y(i+1) - y(i), z(i+1) - z(i)
 	 * Form vector AC(d,e,f) from x(i+2) - x(i), y(i+2) - y(1), z(i+2) - z(i)
 	 */
-	for (i = 0; i < num_mic_locations - 3; i++) {
```

Comment:
Put this into a separate commit. Since this would be a bug fix. Previous change is an optimization for initialize code.

Reply:
btw, I think there's also a typo in the comment in line 158 - it should be `y(i)`, not `y(1)`